src/mirage_kv.ml

```diff
@@ -55,9 +55,9 @@ module type RO = sig
   val pp_error: error Fmt.t
   include Mirage_device.S
   type value
-  val exists: t -> key -> [`Value | `Dictionary] option io
+  val exists: t -> key -> ([`Value | `Dictionary], error) result io
```

Comment:
@linse @hannesm I am very not sure about that change; one would expect `exists` to return some kind of true/false answer, and now it's either `[Value | `Dictionary]` or a `Not_found` error. I think I prefer to have `([Value | Dictionary] option, error) result io` in that case, to signal that `None` means "doesn't exist" but in that case the error could only be stuff private to the implementation (e.g. we can never return any of the the known variants).

WDYT?